HadesScheduler/k8s/job.go

```diff
@@ -0,0 +1,150 @@
+package k8s
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ls1intum/hades/shared/payload"
+	log "github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/client-go/kubernetes"
+)
+
+type K8sJob struct {
+	job              payload.QueuePayload
```

Comment:
In theory, you could use unnamed structs here, which gives you direct access to its members.

I did the same for the DockerJob struct:
```go
type DockerJob struct {
	cli    *client.Client
	logger *slog.Logger
	DockerProps
	payload.QueuePayload // <-- see this line here
}
```